pkgs/development/interpreters/hy/default.nix

```
@@ -1,23 +1,33 @@
-{ stdenv, fetchurl, python2Packages }:
+{ stdenv, python3Packages }:
 
-python2Packages.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "hy";
-  version = "0.17.0";
-
-  src = python2Packages.fetchPypi {
+  version = "0.18.0";
+  src = python3Packages.fetchPypi {
     inherit pname version;
-    sha256 = "1gdbqsirsdxj320wnp7my5awzs1kfs6m4fqmkzbd1zd47qzj0zfi";
+    sha256 = "04dfwm336gw61fmgwikvh0cnxk682p19b4w555wl5d7mlym4rwj2";
   };
 
-  propagatedBuildInputs = with python2Packages; [
+  checkInputs = with python3Packages; [ flake8 pytest ];
+
+  propagatedBuildInputs = with python3Packages; [
     appdirs
     astor
     clint
+    colorama
     fastentrypoints
     funcparserlib
     rply
+    pygments
   ];
 
+  checkPhase = ''
+    PATH=$out/bin:$PATH pytest
+  '';
+
+  # Hy does not include tests in the source distribution from PyPI, so skip testing.
+  doCheck = false;
```

better than nothing, just ensure that it's not trivially broken
```suggestion
  # Hy does not include tests in the source distribution from PyPI, so only test executable.
  checkPhase = ''
    $out/bin/hy --help > /dev/null
  '';
```

---

Ah yeah. So, since doing this, I've found out that doCheck is *usually* false, but I wanted to show that you could run it manually if you pulled the src/git yourself. And, in fact, it runs and passes with what I put in there. I think I understand what you're suggesting, and I agree that it's better than nothing. Since I can make the tests work, but since the tests aren't included in PyPI, what do you think about switching from fetching from PyPI to fetching from github? I would much rather do that and be able to run the real tests rather than just running the simple executable sanity check. I just wasn't sure if we should always prefer PyPI as a "source" when it's available. The same version *is* available from github with the tests, though. Let me know if you think of that route!

---

Ideally there would be unit tests, but some packages I've reviewed only have integration tests, or no tests at all. 

But I would prefer to have unit tests